include/materials/DamagePlasticityStressUpdate.h

```diff
@@ -0,0 +1,183 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#pragma once
+
+#include "MultiParameterPlasticityStressUpdate.h"
+
+class DamagePlasticityStressUpdate : public MultiParameterPlasticityStressUpdate
+{
+public:
+  static InputParameters validParams();
+  DamagePlasticityStressUpdate(const InputParameters & parameters);
+
+  /**
+   * Does the model require the elasticity tensor to be isotropic?
+   */
+  bool requiresIsotropicTensor() override { return true; }
+
+protected:
+  virtual void initQpStatefulProperties() override;
+  virtual void finalizeReturnProcess(const RankTwoTensor & rotation_increment) override;
+
+private:
+  const Real _f_tol;
+  const Real _alfa;
+  const Real _alfa_p;
+  const Real _s0;
+
+  const Real _Chi;
+  const Real _Dt;
+  const Real _ft;
+  const Real _FEt;
+
+  const Real _fyc;
+  const Real _Dc;
+  const Real _fc;
+  const Real _FEc;
+
+  const Real _at;
+  const Real _ac;
+  const Real _zt;
+  const Real _zc;
+  const Real _dPhit;
+  const Real _dPhic;
+  const Real _sqrtPhit_max;
+  const Real _sqrtPhic_max;
+  const Real _dt_bt;
+  const Real _dc_bc;
+  const Real _ft0;
+  const Real _fc0;
+  const Real _small_smoother2;
+
+  const Real _sqrt3;
+
+  /// Whether to provide an estimate of the returned stress, based on perfect plasticity
+  const bool _perfect_guess;
+
+  /// Eigenvectors of the trial stress as a RankTwoTensor, in order to rotate the returned stress back to stress space
+  RankTwoTensor _eigvecs;
+
+
+  MaterialProperty<Real> & _intnl0;
+  MaterialProperty<Real> & _intnl1;
+  MaterialProperty<Real> & _ele_len;
+  MaterialProperty<Real> & _gt;
+  MaterialProperty<Real> & _gc;
+
+  MaterialProperty<Real> & _tD;
+  MaterialProperty<Real> & _cD;
+  MaterialProperty<Real> & _D;
+  MaterialProperty<Real> & _min_ep;
+  MaterialProperty<Real> & _mid_ep;
+  MaterialProperty<Real> & _max_ep;
+  MaterialProperty<Real> & _sigma0;
+  MaterialProperty<Real> & _sigma1;
+  MaterialProperty<Real> & _sigma2;
+
```

Comment:
Add doxygen comments for functions:
/**
 * Get the tensile strength
 * @param intnl Describe this parameter
 * @return tensile strength
 */

Reply:
Addressed ( Need to check InitializeVarV, Trial_stress_params, Stress_params)